pybamm/expression_tree/binary_operators.py

```diff
@@ -229,19 +253,23 @@ class Addition(BinaryOperator):
     A node in the expression tree representing an addition operator.
     """
 
-    def __init__(self, left, right):
+    def __init__(
+        self,
+        left: ChildValue,
+        right: ChildValue,
+    ):
         """See :meth:`pybamm.BinaryOperator.__init__()`."""
         super().__init__("+", left, right)
 
-    def _diff(self, variable):
+    def _diff(self, variable: pybamm.Symbol):
         """See :meth:`pybamm.Symbol._diff()`."""
         return self.left.diff(variable) + self.right.diff(variable)
 
-    def _binary_jac(self, left_jac, right_jac):
+    def _binary_jac(self, left_jac: float | np.ndarray, right_jac: float | np.ndarray):
```

Comment:
And if you define `ChildValue` like I suggest above, this could then be

```suggestion
    def _binary_jac(self, left_jac: ChildValue, right_jac: ChildValue):
```